File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        start = 0
        end = len(nums) - 1

        while start < end :
            sum = nums[start] + nums[end]

            if sum == target:
                return [start+1 , end+1]
            if sum < target:
                start += 1
            else:
                end -= 1
        return [-1,-1]        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int right = numbers.length -1;
        int left = 0;
        int sum = 0;
        int [] res = new int [2];
        while (left < right){
            sum = numbers[right] + numbers[left];
            if (sum == target){
                res[0] = left + 1;
                res[1] = right + 1;
                return res;
            }
            if (sum<target){
                left++;
            }
            else{
                right--;
            }
        }
        res[0] = -1;
        res[1] = -1;
        return res;
        
    }
}